### PR TITLE
bundle gmic

### PR DESCRIPTION
**Pull request by eszlari**

### CODE REVIEW

**Comment by Jehan** [comment]
I am sorry, this won't happen. I have nothing against G'MIC (quite the opposite!) and even work with its developers. But this is the official GIMP flatpak. It is meant to contain GIMP vanilla (as feature-full as possible, and only GIMP core and core plug-ins). We cannot support adding third-party plug-ins here.

Now what can be done is adding an extension point in GIMP (see #22), then you could create a separate flatpak manifest for G'MIC, to be installed through this extension point. You'd be free to maintain this other flatpak yourself (or maybe propose the maintenance to G'MIC developers, though I am not sure at all they have the additional time and would probably appreciate your contribution for maintenance).
So yeah hopefully we will get the extension point soon. That's the current blocker.

In the meantime, sorry but I will close this pull request because I just can't accept the addition. :-/

**Comment by eszlari** [comment]
I have bundled gmic with Krita's flatpak and I can't say, that it increased the maintenance effort (it's only one module more). I know that it is not practical to bundle every plugin out there, but why not the major ones, that most people are using? That's not different from how other flatpaks handle it.

**Comment by Jehan** [comment]
> I have bundled gmic with Krita's flatpak and I can't say, that it increased the maintenance effort (it's only one module more)

Well it's always easier to say when you are not the one who is going to maintain. ;-)

Anyway I don't know about Krita policy and this is not our problem. Our policy is to stick to provide vanilla packages (you'll notice our Windows installer or macOS DMG don't provide G'MIC either). This is the main reason and this already is enough to refuse this request, sorry.

> I know that it is not practical to bundle every plugin out there, but why not the major ones, that most people are using?

And who decides what is major or not? I can tell you for instance that we use GIMP daily professionally here and we don't have the same needs. :-)
Also I don't know if you realize that our flatpak is already quite huge. The more it goes, the bigger it is. Now as the GNOME runtime doesn't even package GTK+2 in particular, this is one more big dependency we have to bundle! So adding now even third-party plug-ins (not used by everyone) is even less a good idea (not that it would have been anyway even without this situation; as I said: vanilla).

> That's not different from how other flatpaks handle it.

Once again, not our problem. We are not here to copy what others are doing.

Anyway I won't budge. This just won't be accepted. Now be assured that I have been seriously looking at this issue in general, in particular these last few months. I am working on better plug-in support in GIMP as a general matter. This includes thinking on how to get **all** plug-ins to be more easily installed and work on **every** platform.
Basically I want generic solutions. Just making exception for every plug-in which someone find nice is not a good long term solution. What about the next time someone comes here with some pull request for some other plug-in, telling me how it is a major one? And then the next-next time with yet another plug-in. These are not theoretical **if**. We have these requests regularly in the main tracker itself to embed more plug-ins into core GIMP, and I can tell everyone has one's own view of what is a "major" plug-in. It's far from an universal concept.

**Comment by eszlari** [comment]
> And who decides what is major or not?

As you know from reddit and twitter, user already said, that missing gmic is a reason for them not to use the flatpak:

https://www.reddit.com/r/GIMP/comments/bbke0h/what_are_the_limitations_of_flatpak_gimp/
https://twitter.com/PedroPS/status/1116079097317351425

Maybe just open a poll on twitter and ask them directly, what plugins they use?

> Also I don't know if you realize that our flatpak is already quite huge.

```
$ flatpak remote-info flathub org.libreoffice.LibreOffice
Download size: 265.8 MB
Installed size: 679.6 MB

$ flatpak remote-info flathub org.freecadweb.FreeCAD
Download size: 248.1 MB
Installed size: 744.1 MB

$ flatpak remote-info flathub org.kde.krita
Download size: 112.7 MB
Installed size: 253.6 MB

$ flatpak remote-info flathub org.gimp.GIMP
Download size: 108.6 MB
Installed size: 307.6 MB
```

> Now as the GNOME runtime doesn't even package GTK+2 in particular, this is one more big dependency we have to bundle!

...which will be solved, when you have ported GIMP to GTK3.

**Comment by Jehan** [comment]
I gave the solution in my first comment for how you can cleanly package G'MIC for GIMP on flatpak.

**Comment by hfiguiere** [comment]
Working on it.
